src/containers/guest-home-page/signup-form/SignupForm.js

```diff
@@ -28,6 +29,14 @@ const SignupForm = ({ handleSubmit, handleChange, handleBlur, data, errors, clos
     setButtonDisabled(!buttonDisabled)
   }
 
```

Comment:
const isValid = useMemo(() => hasErros || noData, [hasErrors, noData])

Reply:
remade through using useMemo. the result is the same as in gif before
once again - we check errors object for undefined values 
and check data object for not empty strings
![Screenshot_2](https://user-images.githubusercontent.com/41526202/216602706-9e841a92-2ad0-46e1-950c-cb531c4247c3.jpg)
so i used .some() for errors object, because its elements added only if error ocurs
and used .every() for data object, because it has 5 elements and every of them has to be not undefined

another variants are cousing unappropriate behavior, and enable user to submit empty form

Reply:
Still we can use the useMemo hook.

Reply:
yes i had used it, refactored a bit